sdk/servicebus/service-bus/src/serializers/ruleResourceSerializer.ts

```diff
@@ -0,0 +1,439 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { HttpOperationResponse } from "@azure/core-http";
+import * as Constants from "../util/constants";
+import {
+  serializeToAtomXmlRequest,
+  deserializeAtomXmlResponse,
+  AtomXmlSerializer
+} from "../util/atomXmlHelper";
+import {
+  getIntegerOrUndefined,
+  getJSONOrUndefined,
+  getStringOrUndefined,
+  getBooleanOrUndefined
+} from "../util/utils";
+import { CorrelationFilter } from "../core/managementClient";
+
+/**
+ * @ignore
+ * Builds the rule options object from the user provided options.
+ * Handles the differences in casing for the property names, converts values to string and ensures the
+ * right order as expected by the service
+ * @param name
+ * @param ruleOptions
+ */
+export function buildRuleOptions(name: string, ruleOptions: RuleOptions = {}): InternalRuleOptions {
+  const internalRuleOptions: InternalRuleOptions = Object.assign({}, ruleOptions, { name: name });
+  return internalRuleOptions;
+}
+
+/**
+ * @ignore
+ * Builds the rule object from the raw json object gotten after deserializing the response
+ * from the service
+ * @param rawRule
+ */
+export function buildRule(rawRule: any): Rule | undefined {
+  if (rawRule == undefined || rawRule == {}) {
+    return undefined;
```

Comment:
In `buildQueue`, `buildTopic`, `buildSubscription`, we check only for `rawData == undefined`, where as here we check for empty object as well. I would suggest to pick one and apply that for all 4 cases

Reply:
Updated